src/common/ceph_context.h

```diff
@@ -143,7 +143,7 @@ class CephContext {
   void do_command(std::string_view command, const cmdmap_t& cmdmap,
 		  std::string_view format, ceph::bufferlist *out);
 
-  static constexpr std::size_t largest_singleton = sizeof(void*) * 72;
+  static constexpr std::size_t largest_singleton = 8 * 72;
```

Comment:
why do we need do replace `sizeof(void*)` with `8` ?

Reply:
sizeof(void*) evaluates differently depending on the bit-ness of the architecture.

  64bit == 8
  32bit == 4
  16bit == 2

so used a literal here ensures that largest singleton always evaluates to the same constant.

Without this change, on 32 bit architectures some of the comparison checking against this value fails as the size is exceeded.

Reply:
@adamemerson do you have any assumption regarding to the `sizeof(void*)` ?

Reply:
No. I picked void* initially thinking in terms of wanting to keep a number that was aligned well, but just switching to 8 should be fine.